pallets/external-validators/src/tests.rs

```diff
@@ -238,7 +244,7 @@ fn duplicate_validator_order_is_preserved() {
             2
         ));
         assert_eq!(ExternalValidators::whitelisted_validators(), vec![3, 1, 2]);
-        assert_ok!(ExternalValidators::set_external_validators(vec![
+        assert_ok!(ExternalValidators::set_external_validators_inner(vec![
             3, 2, 1, 4
         ]));
```

Comment:
maybe it would be worth to test specifically the correctness of events here as a unitest? (i.e., trigger_new_era -> on_era_start)

Reply:
What do you mean? A new test that checks all the events that are emitted on new era? Maybe it's better to have that in typescript, as it's more of an integration test.

Reply:
*disregard*